hooks/useClickOutside.ts

```diff
@@ -0,0 +1,33 @@
+import { useEffect } from 'react';
+import type { RefObject } from 'react';
+
+// eslint-disable-next-line no-unused-vars
+type TypeHandler = (event: MouseEvent | TouchEvent) => void;
+
+type UseClickOutsideProps = {
+  ref: RefObject<HTMLElement>;
+  handler: TypeHandler;
+};
+
+export const useClickOutside = (props: UseClickOutsideProps) => {
+  useEffect(() => {
+    const listener = (event: MouseEvent | TouchEvent) => {
+      if (
```

Comment:
`props.ref.current?.contains(event.currentTarget)`

Reply:
why do that ?

Reply:
If I understand your comment you say my proposal is good ?

Reply:
Both examples are honestly verbose and unclean. Also please @AugustinMauroy do not resolve the conversation without my ack lol

Reply:
```tsx
if (!props.ref.current?.contains(event.currentTarget)) {
  props.handler(event)
}
```

This should do the trick. We don't need double conditions when we have null coalesce operators. Also we should pretty much use `currentTarget` and NOT `target`